src/index.coffee

```diff
@@ -33,13 +32,30 @@ Method.define read, String, "buffer", readBuffer
 stream = require "stream"
 {promise} = require "when"
 
-Method.define read, stream.Readable, (stream) ->
+# Stringifies a stream's buffer according to the given encoding.
+processBuffer = (stream, encoding = "utf8") ->
   buffer = ""
   promise (resolve, reject) ->
-    stream.on "data", (data) -> buffer += data.toString()
+    stream.on "data", (data) -> buffer += data.toString(encoding)
     stream.on "end", -> resolve buffer
     stream.on "error", (error) -> reject error
 
+# Extracts a stream's raw buffer.
+extractBuffer = (stream) ->
+  buffer = ""
+  promise (resolve, reject) ->
+    stream.on "data", (data) -> buffer += data.toString()
+    stream.on "end", -> resolve new Buffer(buffer)
+    stream.on "error", (error) -> reject error
+
+Method.define read, stream.Readable, processBuffer
+Method.define read, stream.Readable, String, processBuffer
+Method.define read, stream.Readable, undefined, extractBuffer
+Method.define read, stream.Readable, "binary", extractBuffer
+Method.define read, stream.Readable, "buffer", extractBuffer
```

Comment:
@dyoder, I modified `read`'s interface to allow returning a raw Buffer while reading a stream.  This allows the stream interface to mirror the file's interface.  

Is this implementation okay, or did you omit that feature to avoid some pitfall?  I opted to create a new `Buffer` instead of using `Buffer::concat` because I was worried concat had more overhead.